compiler/destroyer.nim

```diff
@@ -150,14 +150,16 @@ proc isLastRead(s: PSym; c: var Con; pc, comesFrom: int): int =
       pc = pc + c.g[pc].dest
     of fork:
       # every branch must lead to the last read of the location:
-      let variantA = isLastRead(s, c, pc+1, pc)
+      var variantA = isLastRead(s, c, pc+1, pc)
       if variantA < 0: return -1
       let variantB = isLastRead(s, c, pc + c.g[pc].dest, pc)
       if variantB < 0: return -1
-      pc = variantA+1
+      elif variantA == high(int): 
+        variantA = variantB
+      pc = variantA
```

Comment:
I have added it, because test https://github.com/nim-lang/Nim/blob/devel/tests/destructor/tprevent_assign2.nim fails otherwise. I will have a look in my details why it is happening.

I don't understand how codegen can be involved, could you please explain?